app/ts/simulation/services/EthereumJSONRpcRequestHandler.ts

```diff
@@ -52,6 +52,13 @@ export class EthereumJSONRpcRequestHandler {
 			this.cache.set(hash, responseObject)
 			future.resolve(responseObject)
 			return responseObject
+		} catch(error: unknown) {
+			if (error instanceof Error) {
+				future.reject(error)
+			} else {
+				future.reject(new Error('Unknown error'))
+			}
+			throw error
```

Comment:
Why throw here?  It feels like the future rejection is sufficient?

Reply:
it wasn't sufficient as we never waited for that promise here (it was indended for others to wait it), but I made it more straightforward that it also awaits that